pkg/ebpftracer/c/tracee.bpf.c

```diff
@@ -5442,141 +5443,85 @@ statfunc u32 submit_netflow_event(struct __sk_buff *ctx, net_event_context_t *ne
     e->context.retval = neteventctx->eventctx.retval;
 
     save_to_submit_buf_kernel(&e->args_buf, (void *) &neteventctx->md.flow.proto, sizeof(u8), 0);
-    save_to_submit_buf_kernel(&e->args_buf, (void *) &netflowvalptr->direction, sizeof(u8), 1);
-    save_to_submit_buf_kernel(&e->args_buf, (void *) &neteventctx->md.flow.tuple, sizeof(tuple_t), 2);
-    save_to_submit_buf_kernel(&e->args_buf, (void *) &netflowvalptr->tx_bytes, sizeof(u64), 3);
-    save_to_submit_buf_kernel(&e->args_buf, (void *) &netflowvalptr->rx_bytes, sizeof(u64), 4);
-    save_to_submit_buf_kernel(&e->args_buf, (void *) &netflowvalptr->tx_packets, sizeof(u64), 5);
-    save_to_submit_buf_kernel(&e->args_buf, (void *) &netflowvalptr->rx_packets, sizeof(u64), 6);
+    save_to_submit_buf_kernel(&e->args_buf, (void *) &neteventctx->md.flow.tuple, sizeof(tuple_t), 1);
+    save_to_submit_buf_kernel(&e->args_buf, (void *) &netflowvalptr->tx_bytes, sizeof(u64), 2);
+    save_to_submit_buf_kernel(&e->args_buf, (void *) &netflowvalptr->rx_bytes, sizeof(u64), 3);
+    save_to_submit_buf_kernel(&e->args_buf, (void *) &netflowvalptr->tx_packets, sizeof(u64), 4);
+    save_to_submit_buf_kernel(&e->args_buf, (void *) &netflowvalptr->rx_packets, sizeof(u64), 5);
     net_events_perf_submit(ctx, NET_FLOW_BASE, e);
     return 0;
 }
 
-// Keep track of a flow event if they are enabled and if any policy matched.
-// Submit the flow base event so userland can derive the flow events.
-statfunc u32 cgroup_skb_handle_flow(struct __sk_buff *ctx,
+statfunc u32 cgroup_skb_handle_flow(struct __sk_buff *skb,
                                     net_event_context_t *neteventctx,
-                                    u32 event_type, u32 size, u32 flow)
+                                    u32 event_type, u32 flow)
 {
-    netflowvalue_t *netflowvalptr, netflowvalue = {
-                                       .last_update = bpf_ktime_get_ns(),
-                                       .direction = flow_unknown,
-                                   };
-
     // Set the current netctx task as the flow task.
     neteventctx->md.flow.host_pid = neteventctx->eventctx.task.host_pid;
 
     // Set the flow event type in retval.
     neteventctx->eventctx.retval |= flow;
 
-    // Check if the current packet source is the flow initiator.
-    bool is_initiator = 0;
-    bool ingress = 0;
+    // Drop high port. This is best effort and may not work in all cases.
+    if ((global_config.flow_grouping & FLOW_GROUPING_DROP_SRC_PORT) != 0) {
```

Comment:
If I understand the code correctly, we should be able to determine what port to drop, based on the `ingress` flag in the `neteventctx`. 

Anyway after diving a bit deeper, I guess instead of performing a lookup for the netflow and inverted netflow, we should always base our decisions on the `ingress` flag. This should make the code easier to reason about IMHO.

Reply:
I guess what I am saying is, that we should always transform `flow` into `local -> remote` for `src -> destination`.

Reply:
During TCP connection start we always know who is making a connection and can drop source port. The issue is for samples output. 
But your suggestion makes sense. For samples I can try to lookup by this default condition and if can't find it will try from other end.